firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -176,20 +181,23 @@ public InstallationResponse createFirebaseInstallation(
   private void writeFIDCreateRequestBodyToOutputStream(
       HttpURLConnection httpURLConnection, @NonNull String fid, @NonNull String appId)
       throws IOException {
-    OutputStream outputStream = httpURLConnection.getOutputStream();
-    if (outputStream == null) {
-      throw new IOException(
-          "Cannot send CreateInstallation request to FIS. No OutputStream available.");
-    }
-
-    GZIPOutputStream gzipOutputStream = new GZIPOutputStream(outputStream);
     try {
-      gzipOutputStream.write(
-          buildCreateFirebaseInstallationRequestBody(fid, appId).toString().getBytes("UTF-8"));
+      writeRequestBodyToOutputStream(
+          httpURLConnection, buildCreateFirebaseInstallationRequestBody(fid, appId));
     } catch (JSONException e) {
       throw new IllegalStateException(e);
-    } finally {
-      gzipOutputStream.close();
+    }
+  }
+
+  private static void writeRequestBodyToOutputStream(
+      HttpURLConnection httpURLConnection, JSONObject jsonObject) throws IOException {
+    try (OutputStream outputStream = httpURLConnection.getOutputStream()) {
+      if (outputStream == null) {
+        throw new IOException("Cannot send request to FIS servers. No OutputStream available.");
+      }
+      try (GZIPOutputStream gzipOutputStream = new GZIPOutputStream(outputStream)) {
+        gzipOutputStream.write(jsonObject.toString().getBytes("UTF-8"));
```

Comment:
Please move the calls to .toString().getBytes("UTF-8") outside of this method.
This method should have one "single responsibility":
Writing some content to a stream (and not do transformations on that content).

===============

If you want to remove code duplication for these methods, 
you should name this method appropriately, example"
#writeJsonToOpenConnection()

Even better would be to create a new method with the above name
which does the transformation and then calls this existing method that writes to the given UrlConnection.

Reply:
Done.